### ISSUE TITLE
Add to F-Droid

### ISSUE THREAD

**Issue by prusnak** [nbd-wtf/obw#17]
so people can install OBW easily from F-Droid store.

SBW was added to F-Droid here: https://f-droid.org/en/packages/com.btcontract.wallet/

I can help with the inclusion, but I need the permission of the upstream to include it. What do you think @fiatjaf?

**Comment by prusnak** [comment]
Trying to build from source on F-Droid, the build failed because it cannot find `com.fiatjaf:immortan_2.13:0.7.11-SNAPSHOT`. And indeed the last published version vas 0.7.10 as can be seen here: https://mvnrepository.com/artifact/com.fiatjaf/immortan_2.13

Is it possible to avoid the SNAPSHOT version of the dependencies in the tagged releases?

**Comment by fiatjaf** [comment]
Yes, sorry about that. This snapshot thing wasn't supposed to happen, but I was fixing a bunch of bugs simultaneously and it ended up happening. The next release will be clean as Immortan is more stable.

**Comment by prusnak** [comment]
No worries. I'll patiently wait for the next release. 

About the inclusion: are you OK with me adding OBW to F-Droid? I need explicit approval from you in order to to proceed.

**Comment by fiatjaf** [comment]
I am OK and very grateful for that. I intended to do it but couldn't parse the steps and requirements yet.

But how do we manage releases there? Can I trigger a new release there whenever I make a new release here or do I have to ping you? How does it work?

**Comment by prusnak** [comment]
> Can I trigger a new release there whenever I make a new release here

There are several strategies that can be configured in F-Droid. In my opinion, the best strategy is to let F-Droid automate the release when there is a new tag in the source repository. However, this means that you should never tag an unstable commit, but only commits that you really want to share with F-Droid users as a new update.

**Comment by fiatjaf** [comment]
Yes, let's do that. It will keep me honest and ensure I'll never publish an unbuildable release (like the current) ever again.

**Comment by prusnak** [comment]
> ensure I'll never publish an unbuildable release

I guess nothing bad will happen even if a tag can't be built - I guess F-Droid will just skip this update. But yes, it's always better to tag buildable releases.

**Comment by fiatjaf** [comment]
@prusnak if you're still willing to, there is a new release now and no SNAPSHOT dependencies. It was built just fine on GitHub Actions so it should for you too.

**Comment by prusnak** [comment]
Great!

I created a merge request to fdroid/fdroiddata repository here: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12216

**Comment by prusnak** [comment]
Can we add `changelogs/10.txt` here https://github.com/nbd-wtf/obw/tree/master/fastlane/metadata/android/en-US ?

See https://gitlab.com/-/snippets/1895688 for more info about fastlane.

Original Gitlab comment here: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12216#note_1201349964

For reference, SBW had changelogs here: https://github.com/btcontract/wallet/tree/master/fastlane/metadata/android/en-US/changelogs

**Comment by fiatjaf** [comment]
So, what happens now? Apparently you've fixed all the things they requested on the F-Droid PR, but then the discussion stopped. What can I do to move it forward?

**Comment by prusnak** [comment]
> What can I do to move it forward?

Not sure, I bumped the conversation in the GitLab: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12216#note_1224757130

> So it seems we still haven't resolved the `apksigner` issue, but it seems to me that "Reproducible builds" are just suggested, not required or recommended.
>
> Can we proceed with merging this PR and deal with the `apksigner` later?
